src/borg/repository.py

```diff
@@ -872,7 +879,7 @@ def report_error(msg):
                 continue
             try:
                 objects = list(self.io.iter_objects(segment))
-            except IntegrityError as err:
+            except (IntegrityError, LoggedIO.AtticSegment) as err:
                 report_error(str(err))
                 objects = []
                 if repair:
```

Comment:
If am not 100% sure. Looking at the code, it seems like a repair would convert the Attic segments into proper borg segments. So maybe, we would end up with a semi-converted repository? Or does `borg check` also take care of other places where things need to be converted and perform a full conversion? Either way, this is just backwards-compatible behavior. If this should be fixed, maybe in another commit -- see #2444?